src/Routing/Route/Route.php

```diff
@@ -850,14 +861,22 @@ protected function _writeUrl(array $params, array $pass = [], array $query = [])
      */
     public function staticPath(): string
     {
-        $routeKey = strpos($this->template, '{');
-        if ($routeKey !== false && strpos($this->template, '}') !== false) {
-            return substr($this->template, 0, $routeKey);
+        $matched = preg_match(
```

Comment:
@markstory I replaced the array juggling with a single preg_match. This shouldn't add too much overhead.

Reply:
Much simpler :clap: